File: backend/src/ProductManagement.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;

namespace ProductManagement.Domain.Entities;

public partial class Order
{
    public Guid Id { get; set; }

    public string OrderCode { get; set; } = null!;

    public Guid? UserId { get; set; }

    public string CustomerName { get; set; } = null!;

    public string CustomerPhone { get; set; } = null!;

    public string ShippingAddress { get; set; } = null!;

    public string OrderStatus { get; set; } = null!;

    public string PaymentStatus { get; set; } = null!;

    public string PaymentMethod { get; set; } = null!;

    public decimal Subtotal { get; set; }

    public decimal ShippingFee { get; set; }

    public decimal DiscountAmount { get; set; }

    public decimal TotalAmount { get; set; }

    public Guid? CouponId { get; set; }

    public string? Note { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public virtual Coupon? Coupon { get; set; }

    public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

    public virtual ICollection<OrderStatusHistory> OrderStatusHistories { get; set; } = new List<OrderStatusHistory>();

    public virtual ICollection<ProductReview> ProductReviews { get; set; } = new List<ProductReview>();

    public virtual User? User { get; set; }
}

File: frontend/src/types/order.ts
/**
 * Order Types
 */

export interface OrderItemOption {
  optionId: string;
  optionName: string;
  optionPrice: number;
}

export interface OrderItem {
  productId: string;
  productName?: string;
  quantity: number;
  unitPrice: number;
  itemTotalPrice?: number;
  options?: OrderItemOption[];
}

export interface Order {
  id: string;
  orderCode: string;
  userId: string;
  subtotal: number;
  shippingFee: number;
  discountAmount: number;
  totalAmount: number;
  orderStatus: string;
  paymentStatus: string;
  paymentMethod: string;
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  note?: string;
  createdAt: string;
  items: OrderItem[];
}

export interface CreateOrderItemPayload {
  productId: string;
  quantity: number;
  optionIds?: string[];
}

export interface CreateOrderPayload {
  userId: string;
  items: CreateOrderItemPayload[];
  customerName: string;
  customerPhone: string;
  shippingAddress: string;
  paymentMethod?: string;
  note?: string;
  addressId?: string;
  couponId?: string;
}
